test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+#include <seqan3/io/sequence_file/format_fastq.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/std/filesystem>
+
+using namespace seqan3;
+
+unsigned int const SEED = 1234;
+
+// ============================================================================
+// generate fastq file
+// ============================================================================
+constexpr std::size_t SEQUENCE_LENGTH = 50; //length of nucleotide and quality sequence
+constexpr std::size_t N_ENTRIES_IN_FILE = 4069; // default number of 3-line entries
+
+// if a file of length m is requested once, that file is stored so
+// any functions later on that needs the exact same file can access it
+std::map<std::size_t, std::string> file_dict;
+
+std::string get_file(std::size_t n_entries)
+{
+    bool has_been_init = file_dict.find(n_entries) != file_dict.end();
+
+    if (!has_been_init)
+    {
+        std::cout << std::endl; // sic, helps formatting print output
+
+        std::string file{};
+        std::string const id{"@name"};
+
+        for (size_t i = 0; i < n_entries; ++i)
+        {
+            auto seq = test::generate_sequence<dna5>(SEQUENCE_LENGTH, 0, SEED);
+            std::string seq_string = seq | view::to_char;
+
+            auto quality = test::generate_sequence<phred42>(SEQUENCE_LENGTH, 0, SEED);
+            std::string quality_string = quality | view::to_char;
+
+            file += id + '\n' + seq_string + '\n' + '+' + '\n' + quality_string + '\n';
+        }
+
+        file_dict[n_entries] = file;
+    }
+
+    return file_dict[n_entries];
+}
+
+// ============================================================================
+// write 3-line entry to stream as often as possible
+// ============================================================================
+void fastq_write(benchmark::State & state)
+{
+    std::ostringstream ostream;
+
+    sequence_file_format_fastq format;
+    sequence_file_output_options options{};
+
+    std::string const id{"@name"};
+    auto seq = test::generate_sequence<dna5>(SEQUENCE_LENGTH, 0, SEED);
+    auto quality = test::generate_sequence<phred42>(SEQUENCE_LENGTH, 0, SEED);
+
+    for (auto _ : state)
+    {
+        format.write(ostream, options, seq, id, quality);
+    }
+}
+
+// ============================================================================
+// read dummy fastq file from a stream
+// ============================================================================
+template<std::size_t n_entries_in_file = N_ENTRIES_IN_FILE>
+void fastq_read_from_stream(benchmark::State & state)
+{
+    sequence_file_format_fastq format;
+    sequence_file_input_options<dna5, false> const options{};
+
+    std::string id{};
+    std::vector<dna5> seq{};
+    std::vector<phred42> qual{};
+
+    for (auto _ : state)
+    {
+        std::istringstream istream{get_file(n_entries_in_file)};
```

Comment:
I would move `get_file(n_entries_in_file)` out of the for-loop. AFAIK you try to benchmark reading from a stream.

Reply:
fixed